modules/simple-payments/paypal-express-checkout.js

```diff
@@ -119,37 +142,35 @@ var PaypalExpressCheckout = {
 					env: env
 				};
 
-				return paypal
-					.request
-					.post( PaypalExpressCheckout.getCreatePaymentEndpoint( blogId ), payload )
-					.then( function( paymentResponse ) {
-						return paymentResponse.id;
-					} )
-					.catch( function( paymentError) {
-						PaypalExpressCheckout.showError( 'Item temporarily unavailable', buttonDomId );
-					} );
+				return new paypal.Promise( function( resolve, reject ) {
+					jQuery.post( PaypalExpressCheckout.getCreatePaymentEndpoint( blogId ), payload )
+						.done( function( paymentResponse ) {
+							resolve( paymentResponse.id );
+						} )
+						.fail( function( paymentError ) {
+							var errorMessage = PaypalExpressCheckout.processErrorMessage( paymentError );
+							PaypalExpressCheckout.showError( errorMessage, buttonDomId );
+							reject( new Error( paymentError.responseJSON.code ) );
```

Comment:
I added the error specifically for what Artur noticed in the paypal script. It's useless IMO but it's better than the "Expected reject to be called with Error" bit. I think that is actually coming from the Promise library that the paypal script is using.